crates/task-impls/src/consensus/proposal_helpers.rs

```diff
@@ -975,37 +1009,41 @@ pub async fn handle_quorum_proposal_validated<TYPES: NodeType, I: NodeImplementa
         debug!("decide send succeeded");
     }
 
-    let new_view = task_state.current_proposal.clone().unwrap().view_number + 1;
-    // In future we can use the mempool model where we fetch the proposal if we don't have it, instead of having to wait for it here
-    // This is for the case where we form a QC but have not yet seen the previous proposal ourselves
-    let should_propose = task_state.quorum_membership.get_leader(new_view) == task_state.public_key
-        && consensus.high_qc.view_number
-            == task_state.current_proposal.clone().unwrap().view_number;
-    // todo get rid of this clone
-    let qc = consensus.high_qc.clone();
+    #[cfg(not(feature = "dependency-tasks"))]
```

Comment:
This is a weird looking diff, but everything including line 1045 is feature gated.